modules/geometry-calculation.js

```diff
@@ -1,13 +1,33 @@
-const Line = require('./ecs6-class/line')
+const Line = require('./ecs6-class/line');
+const Point = require('./ecs6-class/point');
+
+
+
 
 const calculateDistance = (point1, point2) => {
+    if (!(point1 instanceof Point) || !(point1 instanceof Point)) {
+        throw Error('The value is of an invalid type')
+    }
     let distanceX = (point2.x - point1.x) ** 2;
-    let distanceY = (point2.y - point2.y) ** 2;
+    let distanceY = (point2.y - point1.y) ** 2;
     const distance = Math.sqrt(distanceX + distanceY);
     return distance;
 }
 
 const calculateJunctionPoint = (line1, line2) => {
+    if (!(line1 instanceof Line) && !(line2 instanceof Line)) {
+        throw Error('The values are of an invalid type')
+    }
+    if (!(line1 instanceof Line)) {
+        throw Error('The first value is of an invalid type')
+    }
+    if (!(line2 instanceof Line)) {
+        throw Error('The second value is of an invalid type')
+    }
+    line1.calculateSlope()
+    line2.calculateSlope()
+    line1.calculateNOfLineFunction()
+    line2.calculateNOfLineFunction()
```

Comment:
the same here, we don't call a function if it is not necessary

Reply:
But the teacher wrote that you should check if the slope and n are correct.
If the user entered wrong values?
and also
If the user entered only n and no slope - the slope will not be calculated.
Therefore, it should be checked separately.